### ISSUE TITLE
Why does pylint connect to the Internet?!

### ISSUE THREAD

**Issue by sam-s** [pylint-dev/pylint#10445]
### Bug description

Pylint reports
```
pip_system_certs: ERROR: truststore not available
```
(I had to disable `truststore` which does not work in my setup)

I guess `pylint` wants to check on packages available using `pip`, who automatically tries to load `truststore`.

Nevertheless, this seems quite wrong, because by calling `pylint` I did _NOT_ authorize internet connections.
I understand that this might seem like an minor peeve, but I urge you to pause and deliberate: would you be comfortable if `gcc` connected to the internet whenever you tried to compile a `C` file? Even if it did not connect, if `ldd /usr/bin/gcc` included an `https` library - would you feel comfortable?

### Command used

```shell
pylint foo.py
```

### Pylint output

```python
pip_system_certs: ERROR: truststore not available
```

### Expected behavior

Between `pylint` and `pip` - nothing related to internet connectivity should be invoked.

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.13.5 (tags/v3.13.5:6cb20a2, Jun 11 2025, 16:15:46) [MSC v.1943 64 bit (AMD64)]
```

### OS / Environment

MSYS_NT-10.0-19045

**Comment by Pierre-Sassoulas** [comment]
I'm not sure how that could happen. Does that output really come from pylint ? Do you have `--unsafe-load-any-extension=y` in your conf ?

**Comment by sam-s** [comment]
> I'm not sure how that could happen. 

actually, this seems to be a _general_ python message when I remove `truststore`

> Does that output really come from pylint ?

it appears when I run `pylint`.

> Do you have `--unsafe-load-any-extension=y` in your conf ?

no
